output/cloud/expv2/output.go

```diff
@@ -19,6 +21,14 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
+// requestMetadatasCollector is an interface for collecting request metadatas
+// and retrieving them, so they can be flushed using a flusher.
+type requestMetadatasCollector interface {
+	CollectRequestMetadatas([]metrics.SampleContainer)
+	PopAll() insights.RequestMetadatas
+}
```

Comment:
nit: I don't particularly see why you need a interface here :shrug: 

I expect this is to facilitate testing only?

Reply:
Yeah, this is currently unused. Do you want me to remove this interface?

Reply:
Yeah, this is currently unused. Do you want me to remove this interface?

Reply:
If it is unused, then we should remove it. I don't see the point in having more abstraction, if not required

Reply:
As discussed in private, it's used in testing. Leaving as is.